### PR TITLE
feat(scoop-cat): Use `bat` to pretty-print JSON

### PR DESCRIPTION
**Pull request by rashil2000**
#### Description

Silly new feature of scoop-cat 🤣

#### Motivation and Context

Your manifest looks cute now xD

##### Before

image

##### After

image

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by niheaven** [comment]
Since 'bat' is somehow slower, use config option such as `CAT_USE_BAT` or `VIEW_USE_BAT` to determine whether using bat even if existed.

Add bat option `--style=grid,numbers` to not show `STDIN` file name. (Have you set `BAT_STYLE`?)

image

**Comment by rashil2000** [comment]
> Since 'bat' is somehow slower,

Will it really matter? We just show one json file at a time, which is hardly 2-3KB.

> use config option such as `CAT_USE_BAT` or `VIEW_USE_BAT` to determine whether using bat even if existed.

I can't find any docs on them. How to use them?

> Add bat option `--style=grid,numbers` to not show `STDIN` file name. (Have you set `BAT_STYLE`?)

Haha yes, I forgot that I had a bat config file. 
Fixed. 
I have used the `-pp` option to keep it visually similar to how it was before.

**Comment by niheaven** [comment]
If it looks the same, why use `bat`? line numbers and grid line are pretty 😄 , but yes, highlighting is the most wonderful one and existed.

For the speed, in my PC (i5-8500/32G), `bat` is slower 20x-30x, i.e., direct output take milliseconds while `bat` take seconds.

> I can't find any docs on them. How to use them?

I mean add new config option to switch the background `scoop cat` uses, i.e., check the config is TRUE and `bat` existed, while the config's default is FALSE.

**Comment by rashil2000** [comment]
> If it looks the same, why use `bat`? line numbers and grid line are pretty 😄 , but yes, highlighting is the most wonderful one and existed. (Hmm, why your terminal doesn't show color? `BAT_STYLE`?)

Grid lines and splits are useful for git output and multiple files. I suppose line numbers are useful, but I don't like them personally. 

My terminal is showing colors (notice the escaped backslashes). I'm using --theme=ansi (because it works in both light and dark terminals).

> For the speed, in my PC (i5-8500/32G), `bat` is slower 20x-30x, i.e., direct output take milliseconds while `bat` take seconds.

> I mean add new config option to switch the background `scoop cat` uses, i.e., check the config is TRUE and `bat` existed, while the config's default is FALSE.

I think I have a genius solution to both these problems: a config variable called `cat_style`. The value of this variable can be:
- **empty**: in this case bat will not be used.
- **not empty**: if set to any of the supported bat styles (see below), it will use bat and pass this value as `--style`.

image

How does that sound?

**Comment by niheaven** [comment]
And `bat` should be installed if `cat_style` is set.

BTW, it should be named `CAT_STYLE`, `cat_style` or `cat-style`? The scoop option names now are a bit messy.

**Comment by rashil2000** [comment]
> And `bat` should be installed if `cat_style` is set.

I'm not sure if we should do this automatically. What if the user doesn't have internet, and just wants to see local manifests? Downloading something for a local functionality seems bad. We can just detect if doesn't exist and use the normal Write-Output in that case. 

> BTW, it should be named `CAT_STYLE`, `cat_style` or `cat-style`? The scoop option names now are a bit messy.

To stick with a convention, I think we should use small case and underscores from now on.

**Comment by niheaven** [comment]
Just like `aria2`, there're two choices. One for automatic enable `bat` unless explicit disabled, while another disable `bat` by default and user could enable it.

For the ones that don't have Internet access, why they want to set `cat_style`? Make notes for this option that it needs Internet.

Treat it just as existed ones, e.g., `aria2` and `lessmsi`. The former is enabled by default, while the latter is enabled by option.

**Comment by rashil2000** [comment]
aria2 and lessmsi are used for downloading and installing apps, which automatically needs internet connection.

I'm against the idea of making an offline functionality require being "online" (even if it's just one time setup).

In the config, I will mention that bat is mandatory if the bat_style option is set (I will also mention in the docs that `This option requires bat. Do 'scoop install bat' to install it.` We also show a similar message for 'sudo' during scoop-checkup.)

**Comment by niheaven** [comment]
Okay, that works fine. Besides, if user has `bat` installed, it should be enabled by default or manually? i.e., the option should be an enable one (`lessmsi`) or disable one (`aria2`)?

**Comment by rashil2000** [comment]
I was initially going for enabled by default, but as you mentioned that bat might be slow on some PCs, we should make it manually enabled.

**Comment by niheaven** [comment]
Okay, please make changes and I'll review again.

**Comment by rashil2000** [comment]
Done

**Comment by niheaven** [comment]
Hmm, could you please separate config underscores to another PR?

**Comment by rashil2000** [comment]
Okay sure

File: lib/install.ps1 [hunk 1]
```diff
@@ -34,9 +34,14 @@ function install_app($app, $architecture, $global, $suggested, $use_cache = $tru
         return
     }
 
-    if ((get_config 'manifest-review' $false) -and ($MyInvocation.ScriptName -notlike '*scoop-update*')) {
-        Write-Output "Manifest: $app.json"
-        Write-Output $manifest | ConvertToPrettyJson
+    if ((get_config 'manifest_review' $false) -and ($MyInvocation.ScriptName -notlike '*scoop-update*')) {
```
Comment (niheaven) [inline lib/install.ps1:37]:
```suggestion
    if ((get_config 'manifest-review' $false) -and ($MyInvocation.ScriptName -notlike '*scoop-update*')) {
```

File: lib/install.ps1 [hunk 2]
```diff
@@ -34,9 +34,14 @@ function install_app($app, $architecture, $global, $suggested, $use_cache = $tru
...
+        "Manifest: $app.json"
```
Comment (niheaven) [inline lib/install.ps1:38]:
```suggestion
        Write-Host "Manifest: $app.json"
```

Do we need this line for pipeline? If not, use `Write-Host`

File: lib/install.ps1 [hunk 3]
```diff
@@ -34,9 +34,14 @@ function install_app($app, $architecture, $global, $suggested, $use_cache = $tru
...
+        $style = get_config cat_style ''
+        if (![String]::IsNullOrWhiteSpace($style)) {
```
Comment (niheaven) [inline lib/install.ps1:40]:
```suggestion
        if (get_config 'cat_style') {
```

image

File: libexec/scoop-cat.ps1 [hunk 4]
```diff
@@ -15,7 +15,12 @@ $app, $bucket, $null = parse_app $app
 $app, $manifest, $bucket, $url = Find-Manifest $app $bucket
 
 if ($manifest) {
-        $manifest | ConvertToPrettyJson | Write-Host
+        $style = get_config cat_style ''
+        if (![String]::IsNullOrWhiteSpace($style)) {
```
Comment (niheaven) [inline libexec/scoop-cat.ps1:19]:
```suggestion
        if (get_config 'cat_style') {
```

See above comment

File: lib/install.ps1 (hunk 1 above)
Comment (rashil2000) [reply lib/install.ps1:37]: This is a bug. The value should be `manifest_review`. See https://github.com/ScoopInstaller/Scoop/blob/d1f828c94275641b86838e90a041af3315fb7d0f/libexec/scoop-config.ps1#L59

File: lib/install.ps1 (hunk 2 above)
Comment (rashil2000) [reply lib/install.ps1:38]: Sure

File: lib/install.ps1 (hunk 3 above)
Comment (rashil2000) [reply lib/install.ps1:40]: $style variable needs to be defined. It is used in the pipeline

File: lib/install.ps1 (hunk 1 above)
Comment (niheaven) [reply lib/install.ps1:37]: Aha, yes

File: lib/install.ps1 (hunk 3 above)
Comment (niheaven) [reply lib/install.ps1:40]: Okay, I'll edit the commit.

File: lib/install.ps1 (hunk 2 above)
Comment (niheaven) [reply lib/install.ps1:38]: See edited comment.

Comment (rashil2000) [reply lib/install.ps1:38]: Write-Host is better, since we don't need to pass it to pipeline

Comment (niheaven) [reply lib/install.ps1:38]: Hmm, I thought you'll use this line... 🤣 

Then `Write-Host`

**Pull request merged**